facad1ng.py

```diff
@@ -57,61 +59,86 @@ def print_banners():
 
 # Input validation functions
 def validate_web_url(url):
-    if not url.startswith("https://") or url.endswith("/"):
-        raise ValueError("Invalid URL format. It should start with 'https://' and not end with '/'.")
+    url_pattern = re.compile(
+    r'^(https://)'  # starts with 'https://'
+    r'([a-zA-Z0-9-]+\.)*'  # optional subdomains
+    r'([a-zA-Z]{2,})'  # domain
+    r'(:\d{1,5})?'  # optional port
+    r'(/.*)?$')
+
+    if not url_pattern.match(url):
+        raise ValueError("Invalid URL format. Please provide a valid web URL.")
 
 def validate_custom_domain(domain):
-    if "://" in domain or "/" in domain:
-        raise ValueError("Invalid custom domain. It should be just the domain name without 'http://' or '/'.")
+    domain_pattern = re.compile(
+        r'^([a-zA-Z0-9]([a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?\.)+[a-zA-Z]{2,}$'
```

Comment:
You can shorten this URL as it takes a lot of steps to match the string. 
from the security perspective, suppose if an attacker provides a huge string, then your regex pattern can take a long time to match the text. Hence, results in either time consumption and unexpected behavior by the regex.